LAPACK_CPP/src/lapack_cpp/rot_cpp.cpp

```diff
@@ -0,0 +1,67 @@
+#ifdef USE_FORTRAN_BLAS
+    #include <cassert>
+    #include <complex>
+    #include <type_traits>
+
+    #include "lapack_c.h"
+    #include "lapack_cpp/base.hpp"
+    #include "lapack_cpp/blas/rot.hpp"
+
+namespace lapack_cpp {
+
+/**
+ * Templated wrapper around c functions, will be instantiated for each type.
+ *
+ * @tparam T
+ * @param A
+ * @param B
+ * @param C
+ */
+template <typename T, typename TC, typename TS, typename idx_t>
+inline void rot_c_wrapper(const Vector<T, idx_t>& x,
+                          const Vector<T, idx_t>& y,
+                          TC c,
+                          TS s)
+{
+    assert(x.size() == y.size());
+
+    if constexpr (std::is_same<T, double>::value) {
+        lapack_c_drot(x.size(), x.ptr(), x.stride(), y.ptr(), y.stride(), c, s);
+    }
+    else if constexpr (std::is_same<T, float>::value) {
+        lapack_c_srot(x.size(), x.ptr(), x.stride(), y.ptr(), y.stride(), c, s);
+    }
+    else if constexpr (std::is_same<T, std::complex<float>>::value) {
+        lapack_c_crot(x.size(), x.ptr(), x.stride(), y.ptr(), y.stride(), c, s);
+    }
+    else if constexpr (std::is_same<T, std::complex<double>>::value) {
+        lapack_c_zrot(x.size(), x.ptr(), x.stride(), y.ptr(), y.stride(), c, s);
+    }
+    else {
+        assert(false);
```

Comment:
The code below fails at compile time:
```c++
static_assert(!std::is_same<T, T>::value, "unknown data type T")
```

References:
* [_How to assert that a constexpr if else clause never happen?_](https://stackoverflow.com/questions/53945490/how-to-assert-that-a-constexpr-if-else-clause-never-happen)